types/eslint/index.d.ts

```diff
@@ -1182,14 +1182,14 @@ export namespace Linter {
     type ParserModule =
         & ESLint.ObjectMetaProperties
         & (
-            | { parse(text: string, options?: any): AST.Program }
+            | { parse(text: string, options?: any): any }
```

Comment:
Personally I don't think this is a solution...

Reply:
👍 `unknown` and `any` in the types are non-starters IMO.